src/parsing/parser.ts

```diff
@@ -188,6 +204,93 @@ export function parseRuleSyntax(
   };
 }
 
+export function parseMappedTrackerSyntax(
+  syntax: MappedTrackerJSON
+): MappedTrackerDefinition {
+  let keyType = syntax.keyType.trim();
+  let valueType = syntax.valueType.trim();
+  var trackerInitialKeys: any[] = [];
+  var trackerInitialValues: any[] = [];
+  for (var pair of syntax.initialvalues) {
+    if (keyType == "uint256") {
+      trackerInitialKeys.push(encodePacked(["uint256"], [BigInt(pair.key)]));
+    } else if (keyType == "address") {
+      const validatedAddress = getAddress(pair.key as string);
+      var address = encodeAbiParameters(parseAbiParameters("address"), [
+        validatedAddress,
+      ]);
+
+      trackerInitialKeys.push(address);
+    } else if (keyType == "bytes") {
+      var bytes = encodeAbiParameters(parseAbiParameters("bytes"), [
+        toHex(stringToBytes(String(pair.key))),
+      ]);
+
+      trackerInitialKeys.push(bytes);
+    } else if (keyType == "bool") {
+      if (pair.key == "true") {
+        trackerInitialKeys.push(encodePacked(["uint256"], [1n]));
+      } else {
+        trackerInitialKeys.push(encodePacked(["uint256"], [0n]));
+      }
+    } else {
+      trackerInitialKeys.push(
+        encodeAbiParameters(parseAbiParameters("string"), [pair.key as string])
+      );
+    }
+
+    if (valueType == "uint256") {
+      trackerInitialValues.push(
+        encodePacked(["uint256"], [BigInt(pair.value)])
+      );
+    } else if (valueType == "address") {
+      const validatedAddress = getAddress(pair.value as string);
+      var address = encodeAbiParameters(parseAbiParameters("address"), [
+        validatedAddress,
+      ]);
+
+      trackerInitialValues.push(address);
+    } else if (valueType == "bytes") {
+      var bytes = encodeAbiParameters(parseAbiParameters("bytes"), [
+        toHex(stringToBytes(String(pair.value))),
+      ]);
+
+      trackerInitialValues.push(bytes);
+    } else if (valueType == "bool") {
+      if (pair.value == "true") {
+        trackerInitialValues.push(encodePacked(["uint256"], [1n]));
+      } else {
+        trackerInitialValues.push(encodePacked(["uint256"], [0n]));
+      }
+    } else {
+      trackerInitialValues.push(
+        encodeAbiParameters(parseAbiParameters("string"), [
+          pair.value as string,
+        ])
+      );
+    }
+  }
+  var keyTypeEnum = 0;
```

Comment:
Could this be updated to `const keyTypeEnum = PT.find(_pt => _pt.name = keyType).enumeration`